### ISSUE TITLE
maven plugin can not refresh Plugin Goals

### ISSUE THREAD

**Issue by SilenceUrStep** [microsoft/vscode-maven#340]
**Describe the bug**
import a project，wait for maven end compile project and like this pic-> 
![image](https://user-images.githubusercontent.com/16930271/59175630-dbbe3680-8b88-11e9-9cb0-a8b4fdf69125.png)

there is no Goals; this happends when i update my vscode,please help!  :[

![image](https://user-images.githubusercontent.com/16930271/59176845-e67aca80-8b8c-11e9-8db3-8c13481fc008.png)

**Comment by Eskibear** [comment]
I myself cannot reproduce it. 

Based on your output, it looks that `help:describe` goal is succesfully executed to fetch detailed info of the plugin. Can you check the content of the temp file (in you log,  after "Wrote description to"), see if the `help:describe` gets the correct output.

**Comment by SilenceUrStep** [comment]
> I myself cannot reproduce it.
> 
> Based on your output, it looks that `help:describe` goal is succesfully executed to fetch detailed info of the plugin. Can you check the content of the temp file (in you log, after "Wrote description to"), see if the `help:describe` gets the correct output.

is empty

**Comment by Eskibear** [comment]
Oh the temp file will be cleaned automatically, sorry about the confusion. 

You may mannually execute the `help:describe` command to see if it gets the correct result. 
The corresponding arguments can be also found in your output channel.
e.g. 
<img width="632" alt="Screen Shot 2019-06-11 at 1 49 03 PM" src="https://user-images.githubusercontent.com/2351748/59246715-be07d480-8c4f-11e9-8649-5af2d80715bb.png">

Then I shall execute following command. (replace the plugin id with your own)
```
mvn help:describe -Dplugin=org.springframework.boot:spring-boot-maven-plugin:2.1.3.RELEASE
```

I get below output
```
Name: Spring Boot Maven Plugin
Description: Spring Boot Maven Plugin
Group Id: org.springframework.boot
Artifact Id: spring-boot-maven-plugin
Version: 2.1.3.RELEASE
Goal Prefix: spring-boot

This plugin has 6 goals:

spring-boot:build-info
  Description: Generate a build-info.properties file based the content of the
    current MavenProject.

spring-boot:help
  Description: Display help information on spring-boot-maven-plugin.
    Call mvn spring-boot:help -Ddetail=true -Dgoal=<goal-name> to display
    parameter details.

......
```

**Comment by SilenceUrStep** [comment]
> Oh the temp file will be cleaned automatically, sorry about the confusion.
> 
> You may mannually execute the `help:describe` command to see if it gets the correct result.
> The corresponding arguments can be also found in your output channel.
> e.g.
> <img alt="Screen Shot 2019-06-11 at 1 49 03 PM" width="632" src="https://user-images.githubusercontent.com/2351748/59246715-be07d480-8c4f-11e9-8649-5af2d80715bb.png">
> 
> Then I shall execute following command. (replace the plugin id with your own)
> 
> ```
> mvn help:describe -Dplugin=org.springframework.boot:spring-boot-maven-plugin:2.1.3.RELEASE
> ```
> 
> I get below output
> 
> ```
> Name: Spring Boot Maven Plugin
> Description: Spring Boot Maven Plugin
> Group Id: org.springframework.boot
> Artifact Id: spring-boot-maven-plugin
> Version: 2.1.3.RELEASE
> Goal Prefix: spring-boot
> 
> This plugin has 6 goals:
> 
> spring-boot:build-info
>   Description: Generate a build-info.properties file based the content of the
>     current MavenProject.
> 
> spring-boot:help
>   Description: Display help information on spring-boot-maven-plugin.
>     Call mvn spring-boot:help -Ddetail=true -Dgoal=<goal-name> to display
>     parameter details.
> 
> ......
> ```
here is what i got，seems the maven plugins command is not correct
![image](https://user-images.githubusercontent.com/16930271/59247468-058f6000-8c52-11e9-8c39-f2c3c277a701.png)

**Comment by SilenceUrStep** [comment]
> Oh the temp file will be cleaned automatically, sorry about the confusion.
> 
> You may mannually execute the `help:describe` command to see if it gets the correct result.
> The corresponding arguments can be also found in your output channel.
> e.g.
> <img alt="Screen Shot 2019-06-11 at 1 49 03 PM" width="632" src="https://user-images.githubusercontent.com/2351748/59246715-be07d480-8c4f-11e9-8649-5af2d80715bb.png">
> 
> Then I shall execute following command. (replace the plugin id with your own)
> 
> ```
> mvn help:describe -Dplugin=org.springframework.boot:spring-boot-maven-plugin:2.1.3.RELEASE
> ```
> 
> I get below output
> 
> ```
> Name: Spring Boot Maven Plugin
> Description: Spring Boot Maven Plugin
> Group Id: org.springframework.boot
> Artifact Id: spring-boot-maven-plugin
> Version: 2.1.3.RELEASE
> Goal Prefix: spring-boot
> 
> This plugin has 6 goals:
> 
> spring-boot:build-info
>   Description: Generate a build-info.properties file based the content of the
>     current MavenProject.
> 
> spring-boot:help
>   Description: Display help information on spring-boot-maven-plugin.
>     Call mvn spring-boot:help -Ddetail=true -Dgoal=<goal-name> to display
>     parameter details.
> 
> ......
> ```

here is what i got, seems the maven plugins command is not correct 
![image](https://user-images.githubusercontent.com/16930271/59247940-8733bd80-8c53-11e9-9ba0-944e24ab355d.png)

**Comment by Eskibear** [comment]
The result looks good... meaning the culprit should be in this extension's code...

* Help-> Toggle Developer Tools, see if there's any suspicious error log.
* Try this simple SpringBoot project, see if the issue still occurs :[test-maven-project.zip](https://github.com/microsoft/vscode-maven/files/3275595/test-maven-project.zip) If the issue still occurs, then it's probably related to specific settings of your machine, you might need to clone the repo and debug it by yourself.

**Comment by SilenceUrStep** [comment]
> 结果看起来不错......意思是罪魁祸首应该在这个扩展的代码中......
> 
> * 帮助 - >切换开发人员工具，查看是否有任何可疑的错误日志。
> * 试试这个简单的SpringBoot项目，看看问题是否仍然存在：[test-maven-project.zip](https://github.com/microsoft/vscode-maven/files/3275595/test-maven-project.zip)如果问题仍然存在，那么它可能与您机器的特定设置有关，您可能需要克隆repo并[自行](https://github.com/microsoft/vscode-maven/files/3275595/test-maven-project.zip)调试。

i try ur project , but there's no problem at all, show u the picture
![image](https://user-images.githubusercontent.com/16930271/59260810-dfc48400-8c6e-11e9-80dc-747f041ceaba.png)

and i use Developer Tools,got some error log , show u the picture
![image](https://user-images.githubusercontent.com/16930271/59261237-b1937400-8c6f-11e9-8541-ca9d88a776f0.png)

here is the log 
```
(node:1268) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
t.log @ /C:/code_env/Microsoft VS Code/resources/app/out/vs/workbench/workbench.main.js:238
```

**Comment by Eskibear** [comment]
So it only occurs for your project. You may try the following:
**Right Click** on the plugin(e.g. the jetty plugin) -> Click **"Refresh**", see if it gets back to work.
If not, can you share your project with us?

BTW, the DeprecationWarning in the log is known to have nothing to do with this issue.

**Comment by andreashouben** [comment]
I have the same issue with our project. Because I cannot share it, I created a small demo project which shows the same issue.

<img width="520" alt="grafik" src="https://user-images.githubusercontent.com/3708288/61136397-daec2d80-a4c3-11e9-9cd0-1a74a61ebf66.png">
I attached it as .zip.
[demo.zip](https://github.com/microsoft/vscode-maven/files/3386826/demo.zip)

**Comment by andreashouben** [comment]
Running `mvn help` generates output though:
```
mvn help:describe -Dplugin=org.apache.maven.plugins:maven-compiler-plugin:3.8.0                                                                                                          10211  16:42:58
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- maven-help-plugin:3.2.0:describe (default-cli) @ standalone-pom ---
[INFO] org.apache.maven.plugins:maven-compiler-plugin:3.8.0

Name: Apache Maven Compiler Plugin
Description: The Compiler Plugin is used to compile the sources of your
  project.
Group Id: org.apache.maven.plugins
Artifact Id: maven-compiler-plugin
Version: 3.8.0
Goal Prefix: compiler

This plugin has 3 goals:

compiler:compile
  Description: Compiles application sources

compiler:help
  Description: Display help information on maven-compiler-plugin.
    Call mvn compiler:help -Ddetail=true -Dgoal=<goal-name> to display
    parameter details.

compiler:testCompile
  Description: Compiles application test sources.

For more information, run 'mvn help:describe [...] -Ddetail'

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.360 s
[INFO] Finished at: 2019-07-12T16:43:31+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by Eskibear** [comment]
@andreashouben Thank you for providing the project, and I can reproduce it. 
After the very initial investigation, I find it's probably because of unexpected ANSI escape code in the output. E.g. `ESC[m` 

![image](https://user-images.githubusercontent.com/2351748/61172751-cab77980-a5bb-11e9-93f5-0eaed307d606.png)

I'll open a new issue to track this.

**Comment by Eskibear** [comment]
@andreashouben @SilenceUrStep Can you have a try with the latest build below?
[vscode-maven.zip](https://github.com/microsoft/vscode-maven/files/3390768/vscode-maven.zip)

**Comment by SilenceUrStep** [comment]
![image](https://user-images.githubusercontent.com/16930271/61194529-a317fc80-a6f4-11e9-9fe3-60d393dc2700.png)
i've try ur new plugin，it works，many thanks!

**Comment by andreashouben** [comment]
<img width="490" alt="grafik" src="https://user-images.githubusercontent.com/3708288/61196227-579e2800-a6cd-11e9-8310-090b37105af1.png">
Yes! Now it works :-) I do have the issue though that the extended entries tend to collapse after loading. But that's just a minor issue. Thank you!

**Comment by Eskibear** [comment]
Thank you for verification, I'm closing it now. And the fix should be released soon, in v0.18.1

> I do have the issue though that the extended entries tend to collapse after loading. But that's just a minor issue. 

I've created a new issue to track that.
